.github/scripts/trymerge.py

```diff
@@ -847,12 +847,17 @@ def checks_to_str(checks: List[Tuple[str, Optional[str]]]) -> str:
     raise RuntimeError(reject_reason)
 
 
-def get_pending_checks(pr: GitHubPR) -> List[Tuple[str, str]]:
+def pr_get_checks_with_lambda(pr: GitHubPR, status_check: Callable[[Optional[str]], bool]) -> List[Tuple[str, str]]:
     checks = pr.get_checkrun_conclusions()
-    return [(name, status[1]) for name, status in checks.items() if status[0] is None]
+    return [(name, status[1]) for name, status in checks.items() if status_check(status[0])]
 
-def pr_has_pending_checks(pr: GitHubPR) -> bool:
-    return len(get_pending_checks(pr)) > 0
+
+def pr_get_pending_checks(pr: GitHubPR) -> List[Tuple[str, str]]:
+    return pr_get_checks_with_lambda(pr, lambda x: x is None)
+
+
+def pr_get_failed_checks(pr: GitHubPR) -> List[Tuple[str, str]]:
+    return pr_get_checks_with_lambda(pr, lambda x: x == "FAILURE")
```

Comment:
I think we also want to check if CANCELLED?

Reply:
We need to refine and test if further, IMO. I often see "CANCELLED" reported as "SKIPPED", but we always have one legitimate "SKIPPED" job on PR (TorchBench)